client/asset/btc/btc.go

```diff
@@ -4246,17 +4251,89 @@ func (btc *baseWallet) DepositAddress() (string, error) {
 	if btc.node.locked() {
 		return addrStr, nil
 	}
+
 	// If the wallet is unlocked, be extra cautious and ensure the wallet gave
 	// us an address for which we can retrieve the private keys, regardless of
 	// what ownsAddress would say.
-	priv, err := btc.node.privKeyForAddress(addrStr)
+	priv, err := btc.getPrivKeyForAddress(addrStr)
 	if err != nil {
 		return "", fmt.Errorf("private key unavailable for address %v: %w", addrStr, err)
 	}
 	priv.Zero()
 	return addrStr, nil
 }
 
+// getPrivKeyForAddress accomodates `dumpprivkey' RPC for different RPC sequencing
+func (btc *baseWallet) getPrivKeyForAddress(addr string) (*btcec.PrivateKey, error) {
+	if btc.walletInfo.Name == "Firo" {
+		return btc.getPrivkeyFiro(addr)
+	}
+	return btc.node.privKeyForAddress(addr)
+}
```

Comment:
I think the preferred way to override functionality is like this: https://github.com/decred/dcrdex/commit/612e57b7a31093e7f23bf7c926d74a9bfc92ab4c
At least this is the direction @buck54321 was going with the last zec update, so maybe he can confirm.